src/main/java/application/rest/v1/actions/AppPodlistFunction.java

```diff
@@ -0,0 +1,162 @@
+
+// apppodlist() or apppodlist(<application-namespace>,<application-name>)
+// Returns a list of pod names from all deployments belonging to specified application.
+// Return value structure is: { "pods" : "[<name1>,<name2>,etc]" }
+public class AppPodlistFunction implements Function {
+    
+    private static final String PODS_PROPERTY_NAME = "pods";
+    private static final String DEPLOYMENT_KIND = "Deployment";
+    private static final String APPLICATION_KIND = "Application";
+    private static final String POD_KIND = "Pod";
+    private static final String METADATA_PROPERTY_NAME = "metadata";
+
+    @Override
+    public String getName() {
+        return "apppodlist";
+    }
+
+    @Override
+    public boolean allowedParameterCount(int parameterCount) {
+        return parameterCount == 0 || parameterCount == 2;
+    }
+
+    @Override
+    public String invoke(ResolutionContext context, List<String> parameters) {
+        
+        final ApiClient client = context.getApiClient();
+        final ComponentInfoRegistry registry = context.getComponentInfoRegistry();
+        final JsonObject resource; 
+        String result = null;     
+
+        if (parameters.size() == 0) {
+            if (!APPLICATION_KIND.equals(context.getResourceKind())) {
+                // The context resource isn't a deployment.
+                return null;
+            }
+            resource = context.getResource();
+        }
+        // Two parameter function
+        else {
+            // parameters.get(0) :: appNamespace
+            // parameters.get(1) :: appName
+            final String appNamespace = parameters.get(0);
+            final String appName = parameters.get(1);
+            try {
+                //get application namespaced object
+                Object o = registry.getNamespacedObject(client, APPLICATION_KIND, appNamespace, appName);
+                resource = KAppNavEndpoint.getItemAsObject(client, o);      
+                result = getPodListFromApp(client, resource, registry);         
+            }
+            catch (ApiException e) {
+                return null;
+            }  
+        }
+        return result;                   
+    }
+
+
+    private String getPodListFromApp(ApiClient client, JsonObject resource, ComponentInfoRegistry registry) {
+        PodlistResult result = new PodlistResult();       
+        final Selector aSelector = Selector.getSelector(resource);                
+        if (!aSelector.isEmpty()) {
+            final String alabelSelector = aSelector.toString();           
+            try {  
+                //retrieve deployment object from app with the labels matching 
+                Object deplO = registry.listClusterObject(client, DEPLOYMENT_KIND, null, alabelSelector, null, null);
+                List<JsonObject> deployments = KAppNavEndpoint.getItemsAsList(client, deplO); 
+                deployments.forEach(d -> {
+                    String deplName = null;
```

Comment:
Not sure why line 98 to 116 is necessary. The forEach() is iterating over the deployments. Should be able to extract the selector from each of them directly (e.g. Selector dSelector = Selector.getSelector(d)).